sat/cli/bootsys/mgmt_power.py

```diff
@@ -280,11 +281,18 @@ def do_mgmt_shutdown_power(ssh_client, username, password, excluded_ncns, ncn_sh
             start_shutdown(other_ncns, ssh_client)
             LOGGER.info(f'Waiting up to {ncn_shutdown_timeout} seconds for other NCNs to '
                         f'reach powered off state according to ipmitool: {", ".join(other_ncns)}.')
-            finish_shutdown(other_ncns, username, password,
-                            ncn_shutdown_timeout, ipmi_timeout)
-            LOGGER.info('Shutdown and power off of all other NCNs complete.')
+
+            # Check if the user wants to proceed with hard power off after timeout
+            prompt_message = 'Do you want to forcibly power off the nodes after timeout?'
+            if not hard_power_off(prompt_message):
```

Comment:
The function `hard_power_off` always returns either the string `'yes'` or `'no'`, so this will always be false.

There actually exists a function `pester_choices` in `sat.util` that would work here. Or possibly `prompt_continue` in that same module. If you used `pester_choices`, it would look something like:

```suggestion
            response = pester_choices('forced power off of management NCNs', ('yes', 'no'))
            if response != 'yes':
```

If you used `prompt_continue`, it would just be:

```suggestion
            prompt_continue('forced power off of management NCNs')
```

Using `prompt_continue` would result in an exit code of 1, which is arguably okay, but different than what you wrote.